### PR TITLE
Fix spire-server configmap UpstreamAuthority/aws_pca and KeyManager/a…

### PR DESCRIPTION
**Pull request by unufr33**
Current configmap template renders to a wrong KeyManager and UpstreamAuthority configurarion when aws_kms and aws_pca are enabled and container is crashing. The proposed changes will fix the issue.

### CODE REVIEW

**Comment by kfox1111** [comment]
Thanks for the fixes.

Could you please add a quick section to tests/unit/spire_test.go that turns the features on to see if it renders ok? Not having any tests at all, even "does it render to valid yaml" makes it hard to not accidentally break.

**Comment by faisal-memon** [comment]
@unufr33 Do you have time to write the test? No problem if not, we can merge without it. We understand people are busy.

**Comment by unufr33** [comment]
Hope this works

File: tests/unit/spire_test.go [hunk 1]
```diff
@@ -71,4 +71,38 @@ spire-server:
...
+      plugin_data: {}
+    disk:
+      enabled: false
+`)
+			Expect(err).Should(Succeed())
+			notes := objs["spire/charts/spire-server/templates/configmap.yaml"]
+			Expect(notes).Should(ContainSubstring("\"aws_kms\": {"))
+        })
+
+	})
```
Comment (faisal-memon) [inline tests/unit/spire_test.go:91]:
```suggestion
        })
	})
```

File: tests/unit/spire_test.go [hunk 2]
```diff
@@ -71,4 +71,38 @@ spire-server:
...
+      enabled: true
+      region: us-west-2
+      plugin_data: {}
+`)
+			Expect(err).Should(Succeed())
+			notes := objs["spire/charts/spire-server/templates/configmap.yaml"]
+			Expect(notes).Should(ContainSubstring("\"aws_pca\": {"))
+        })
+
+	})
```
Comment (faisal-memon) [inline tests/unit/spire_test.go:107]:
```suggestion
        })
	})
```

File: tests/unit/spire_test.go [hunk 3]
```diff
@@ -71,4 +71,36 @@ spire-server:
...
+      enabled: true
+      region: us-west-2
+      plugin_data: {}
+    disk:
+      enabled: false
+`)
+			Expect(err).Should(Succeed())
+			notes := objs["spire/charts/spire-server/templates/configmap.yaml"]
+			Expect(notes).Should(ContainSubstring("\"aws_kms\": {"))
+        })
```
Comment (faisal-memon) [inline tests/unit/spire_test.go:89]: The spacing seems off in this file. Can you try running `go fmt -w tests/unit/spire-test.go` on it?

Comment (unufr33) [reply tests/unit/spire_test.go]: done

**Comment by faisal-memon** [comment]
Is this test failure related to this pr?

```
INFO: Downloading bootstrap version 'v2.0.0' of cosign to verify version to be installed...
      https://storage.googleapis.com/cosign-releases/v2.0.0/cosign-linux-amd64
Error: Process completed with exit code 56.
```

**Comment by unufr33** [comment]
I don't think the failed tests are related to my changes. My changes are in the "checks" job and it passes. The failed jobs passed several times before that .Can you please rerun ?

**Comment by kfox1111** [comment]
> Is this test failure related to this pr?
> 
> ```
> INFO: Downloading bootstrap version 'v2.0.0' of cosign to verify version to be installed...
>       https://storage.googleapis.com/cosign-releases/v2.0.0/cosign-linux-amd64
> Error: Process completed with exit code 56.
> ```

Looks like a transient network issue. I just re-triggered the failed job to rerun. I'm expecting it to pass.

**Comment by kfox1111** [comment]
> I don't think the failed tests are related to my changes. My changes are in the "checks" job and it passes. The failed jobs passed several times before that .Can you please rerun ?

Yeah. The tests look great. Thanks for writing them. :)

**Review by kfox1111 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**